### ISSUE TITLE
p2p system doesn't gossip hotspot information correct. AWS nodes still trying to connect to different ports other than 44158

### ISSUE THREAD

**Issue by axkman** [helium/miner#1171]
Hi!

I've the problem that since 23sept my hotspot doesn't beacon or challenges etc. It witnesses other hotspots fine and both receiving and transmitting RF side works great (great setup,location).
Since 22 sept I moved my hotspot to another location and thats where the problems start. Internet connection very okay, port forwarded. Everything works.

Except for the p2p network that gets confused when IP changes. The p2p seems to have troubling updating/gossipping new information.
A few hotspots get the updated information by p2p gossip. But there is still some parts of the p2p network (aws nodes for example) that try to connect to ip/ports based on old hotspot information.
I now still get aws nodes and hotspots trying to connect to my hotspot on different ports while they should be connecting to its ip and port 44158, they had since 23sept 2021 to update/gossip the correct information, yet aws nodes and hotspots still trying to connect to my hotspot on other ports than 44158.
I suspect that my hotspot shows offline regularly on explorer/helium app, because part of the p2p network still thinks my hotspot is on other ip or different port than 44158 and when it finds it cannot connect to it at those (other than 44158) ports, it makes the conclusion "oh the hotspot is offline/inactive, lets gossip the offline/inactive information and lets not instruct the hotspot to send out beacons and other activities".

Same conclusion is when I try to mess with the ports, when I force my hotspot to be relayed, then after some hours different information about it is gossiped thru the p2p network, but then even then, it still varies its syncing/synced/offline status in explorer/helium app. So that means that old hotspot status information is still present instead of the new current hotspot state.
Its like various parts of the p2p network has different hotspot state information present, not being updated to the hotspots current state.

When I then open port 44158 again, reboot hotspot and router etc. Then it takes a while to update it current state (which is normal I suppose) then a few days after I check my router logs/explorer/helium app and notice that still hotspots/aws nodes try to connect to its old ports. While the p2p network had the time to update the hotspot state to its current state. Parts of the p2p network failed to update to current state and keep holding on to old information. That information I believe cause the (actually being online and port 44158 fwd'd etc) hotspot to show offline in explorer/helium app and cause it not to perform activities like beaconing/challenges etc.
[https://postimg.cc/YvbcvJQg](url)

**Comment by ke6jjj** [comment]
Are you certain that the nodes are trying to connect to _you_? That's not how seed nodes work.

Your miner connects outbound _to_ the seed nodes. They do indeed provide p2p services on a port other than TCP 44158, so that's why you see those port numbers in your traffic analysis.

**Comment by axkman** [comment]
Yeah they connected towards my hotspot - so inbound traffic. My router logs only shows firewall inbound blocked traffic. So the entries i saw are only inbound and blocked. The outbound i don't see.
So a lot of hotspot and awsnodes :443 were connected to my hotspot on all different kind of ports.

I have to say that after the 24hours blockchain halt, things got better for my hotspot. Its like that the current ip/port information was gossiped well thru the p2p network again and my hotspot started to beacon again aswell.
I think it has also something to do that in the recent firmwares after the 24hour blockchain halt there were quite some firmwares with updates but also related to p2p gossip. That might have fixed it.

**Comment by abhay** [comment]
This report is somewhat old so I'm going to close it for now. Hopefully you've seen some improvement since the more recent releases. Please verify that your Hotspot is updated to the latest version, wait a few days to see if activity has resumed and then, if not, please post in the pinned issue with more details on what you're seeing.
